dj_cqrs/signals.py

```diff
@@ -62,7 +66,16 @@ def post_save(cls, sender, **kwargs):
         queue = kwargs.get('queue', None)
 
         connection = transaction.get_connection(using)
-        if not connection.in_atomic_block:
+        if not connection.in_atomic_block or instance.is_initial_cqrs_save:
+            transaction.on_commit(
+                lambda: cls._post_save_produce(sender, instance, using, sync, queue),
```

Comment:
if not in transaction, post_save_produce() will be called instantly - same behaviour as before